src/textual/_types.py

```diff
@@ -28,3 +28,4 @@ def post_message_no_wait(self, message: "Message") -> bool:
 
 SegmentLines = List[List["Segment"]]
 CallbackType = Union[Callable[[], Awaitable[None]], Callable[[], None]]
+WatchCallbackType = Union[Callable[[Any], Awaitable[None]], Callable[[Any], None]]
```

Comment:
I'm still not familiarised with how `ParamSpec` works, but wouldn't this be the type of place where it makes sense, to somehow tie the signature of the function that is the callback to the place where it gets called?

Reply:
I don't think it would work here. At the point where you add watchers, it is too dynamic to type. You do something like `self.watch(app, "dark", on_dark_change)`, but the typing information isn't there at that point. You can only know the type of "dark" at runtime.

Reply:
I see where you're going. Thanks!